### PR TITLE
Game logic quiz

### LINKED ISSUE #42: Change export syntax of storyEnd

### LINKED ISSUE #48: Export quizComponent
In the `quiz-component.js` file, quizComponent isn't exported.

### PR DESCRIPTION
**Pull request by mdlugosz1**
Here is extend game logic that shows modules in right order and handles quiz component. Sorry in advance for a lot of commits, but i made some changes in other files to make things work properly. I wrote comments in those other files which includes my thoughts about why i changed something. Please read them and let me know what you think about these changes. We can always take another way to solve some problems :)

### CODE REVIEW

File: src/js/chapterDOM.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
 import { makeDraggable, makeDroppable } from "./drag-drop";
 
 const createChapterStructure = (chapterObj, callback) => {
+  /* This method will show images and question and hide chapter story when player
+  clicks next btn on chapter story page */
+  const hideStoryConent = () => {
```
Comment (Tudwall) [inline src/js/chapterDOM.js]: I think you meant hideStoryContent, no?

**Comment by Tudwall** [comment]
No worries about the amount of commits, it's not a bad thing to me

File: src/js/chapterDOM.js (hunk 1 above)
Comment (mdlugosz1) [reply src/js/chapterDOM.js]: Yeah, wrong comment placement.

File: src/js/story-factory.js [hunk 2]
```diff
@@ -1,31 +1,48 @@
-const Story = (title) => {
+/* Removed findFirstChapter. Added chapters sorting in addChapters method. I added getCurrentChapter which is more felxible because find method
+always returns first element that meets given requirements so in every new story this element is first chapter (thanks to sort method) and it 
+also finds first chapter which completion status is false.
+
+I added quiz parameter since i realized how and when quizes should be displayed, then it makes sense to pass all story quizzes when we initialize
+new story object. Also, like i wrote in game logic - getFinalQuizzes probably shold be changed to method similar to getCurrentChapter.
+ */
```
Comment (DavidMatalik) [inline src/js/story-factory.js:7]: Alright yeah changes of how to get chapters make sense to me. (Haven't checked it in detail though)

`getFinalQuizzes` for sure needs to be somehow changed. @emdevs wrote the QuizSummary factory in her last PR which was merged recently. I think you can make use of that to get all the quizzes of the story and if needed add there a method to get the current quiz.

**Review by DavidMatalik — APPROVED** [review]
Code looks good to me. Tested game and it works quite well.

File: src/js/chapterDOM.js (hunk 1 above)
Comment (Tudwall) [reply src/js/chapterDOM.js]: I meant the typo haha

Comment (mdlugosz1) [reply src/js/chapterDOM.js]: Haha, now i realized that there is a typo. Definitely it should be content instead of conent :D

**Review by Tudwall — APPROVED** [review]

**Review by emdevs — COMMENTED** [review]
Looks good to me! Thank you for writing the comments, it made the changes easier to follow.

**Review by emdevs — APPROVED** [review]

**Pull request merged**